src/rows.js

```diff
@@ -52,17 +53,20 @@ export class Rows {
     if (this._done) return { done: true };
     // Load row data and advance statement
     const row = this._get();
-    switch (this._db._wasm.step(this._db._id, this._id)) {
+    const status = this._db._wasm.step(this._db._id, this._id);
+    switch (status) {
       case constants.status.sqliteRow:
         // NO OP
         break;
       case constants.status.sqliteDone:
         this.done();
         break;
       default:
-        // TODO: Make more helpful
         this.done();
-        throw new Error("Internal error.");
+        throw new SqliteError(
```

Comment:
This won't work as expected (you need to reference the db's id, not the statement's id). I think the `_error` function of the db object should be used here (it's used in DB itself whenever an error is generated based on a returned code):

```suggestion
        throw this._db._error(status);
```

Reply:
My bad! :sweat_smile:
Fixed.